06-classification_continued.Rmd

```diff
@@ -194,18 +197,100 @@ cancer_test_predictions %>%
+expert diagnosis. This is why it is important not only to look at accuracy, but
+also the confusion matrix.
+
+However, there is always an easy baseline that you can compare to for any
+classification problem: the *majority classifier*. The majority classifier
+*always* guesses the majority class label from the training data, regardless of
+what values the predictor variables take.  It helps to give you a sense for
+scale when considering accuracies. If the majority classifier obtains a 90%
+accuracy on a problem, then you might hope for your K-nearest neighbours
+classifier to do better than that. If your classifier provides a significant
+improvement upon the majority classifier, this means that at least your method
+is extracting some useful information from your predictor variables.  Be
+careful though: improving on the majority classifier does not *necessarily*
+mean the classifier is working well enough for your application. 
+
+As an example, in the breast cancer data, the proportions of benign and malignant
+observations in the training data are as follows:
+```{r 06-proportions}
+cancer_proportions <- cancer_train %>%
+                      group_by(Class) %>%
+                      summarize(n = n()) %>%
+                      mutate(percent = 100*n/nrow(cancer_train))
+cancer_proportions
+```
+```{r 06-proportions-2, echo = FALSE, warning = FALSE}
+cancer_propn_1 <- cancer_proportions %>%
+                filter(Class == 'B') %>%
+                select(percent)
 ```
 
-This says that the classifier labelled 43+82 = 125 observations correctly,
- 10 observations as benign when they were truly malignant,
-and 7 observations as malignant when they were truly benign. 
+Since the benign class represents the majority of the training data,
+the majority classifier would *always* predict that a new observation
+is benign. The estimated accuracy of the majority classifier is usually
+fairly close to the majority class proportion in the training data.
+In this case, we would suspect that the majority classifier will have 
+an accuracy of around `r round(cancer_propn_1[1,1], 0)`%.
+The K-nearest neighbours classifier we built does quite a bit better than this, 
+with an accuracy of `r round(100*cancer_acc_1$.estimate, 0)`%. 
+This means that from the perspective of accuracy,
+the K-nearest neighbours classifier improved quite a bit on the basic
+majority classifier. Hooray! But we still need to be cautious; in 
+this application, it is likely very important not to miss-diagnose any malignant tumours to avoid missing
+patients who actually need medical care. The confusion matrix above shows
+that the classifier does indeed miss-diagnose a non-negligible number malignant tumours as benign.
+Therefore, even though the accuracy improved upon the majority classifier,
+our critical analysis suggests that this classifier may not have appropriate performance
+for the application.
 
 ## Tuning the classifier
 
```

Comment:
I want to avoid introducing a bunch of methodology/text/code that ends up not being used by students because "they should do this other thing". I will try to rephrase this bit to make it take less of a leap